src/Rector/Deprecation/DrupalServiceRenameRector.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace DrupalRector\Rector\Deprecation;
+
+use DrupalRector\Rector\ValueObject\DrupalServiceRenameConfiguration;
+use PhpParser\Node;
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+class DrupalServiceRenameRector extends AbstractRector implements ConfigurableRectorInterface {
+
+    /**
+     * @var DrupalServiceRenameConfiguration[] $staticArgumentRenameConfigs
+     */
+    protected array $staticArgumentRenameConfigs = [];
+
+    public function configure(array $configuration): void {
+        foreach ($configuration as $value) {
+            if (!($value instanceof DrupalServiceRenameConfiguration)) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Each configuration item must be an instance of "%s"',
+                    DrupalServiceRenameConfiguration::class
+                ));
+            }
+        }
+
+        $this->staticArgumentRenameConfigs = $configuration;
+    }
+
+    public function getNodeTypes(): array {
+        return [
+            Node\Expr\StaticCall::class,
+        ];
+    }
+
+    public function refactor(Node $node) {
+        if ($node instanceof Node\Expr\StaticCall) {
+            foreach ($this->staticArgumentRenameConfigs as $configuration) {
+                if ($this->getName($node->name) === 'service' && (string) $node->class === 'Drupal') {
+                    if (count($node->args) === 1) {
+                        /* @var Node\Arg $argument */
+                        $argument = $node->args[0];
```

Comment:
🤔 I wonder if we can also detect usages of `ContainerInterface::get` here?

We'd need PHPStan's scope, probably. So if `$node->name === 'get'` but only if the class/object is a container object. Maybe as a follow up ticket to this PR so we can land the refactor? Makes it more resilient for any future changes like this.

Reply:
Hmm, we might actually be able to do that yeah. This would also enable refactoring non-global calls on this which could be great.

I agree on a follow up though, or perhaps somehting we can explore when we start writing rules again.

Reply:
https://www.drupal.org/project/rector/issues/3382531